From 870b2d2c41328092447540d4aae52c12ac1e1fcc Mon Sep 17 00:00:00 2001
From: Mohan R <mohan43u@gmail.com>
Date: Sat, 8 Aug 2015 04:22:00 +0000
Subject: git: track libgit2/libgit2-glib ABI breakages

https://bugzilla.gnome.org/show_bug.cgi?id=753375

#diff --git a/configure.ac b/configure.ac
#index 54062c0..a81b5f3 100644
#--- a/configure.ac
#+++ b/configure.ac
#@@ -145,7 +145,7 @@ dnl ***********************************************************************
# m4_define([gtk_required_version], [3.17.1])
# m4_define([glib_required_version], [2.45.1])
# m4_define([gtksourceview_required_version], [3.17.3])
#-m4_define([ggit_required_version], [0.22.9])
#+m4_define([ggit_required_version], [0.23.4])
# m4_define([gjs_required_version], [1.42.0])
# m4_define([pygobject_required_version], [3.0.0])
# m4_define([devhelp_required_version], [3.16.0])
diff --git a/src/dialogs/gb-new-project-dialog.c b/src/dialogs/gb-new-project-dialog.c
index dfa7e91..8f205a9 100644
--- a/src/dialogs/gb-new-project-dialog.c
+++ b/src/dialogs/gb-new-project-dialog.c
@@ -211,12 +211,12 @@ gb_new_project_dialog__clone_worker (GTask        *task,
   ggit_clone_options_set_is_bare (clone_options, FALSE);
   ggit_clone_options_set_checkout_branch (clone_options, "master");
   ggit_clone_options_set_fetch_options (clone_options, fetch_options);
-  ggit_fetch_options_free (fetch_options);
+  g_clear_pointer (&fetch_options, ggit_fetch_options_free);
 
   repository = ggit_repository_clone (req->uri, req->location, clone_options, &error);
 
-  g_object_unref (callbacks);
-  ggit_clone_options_free (clone_options);
+  g_clear_object (&callbacks);
+  g_clear_object (&clone_options);
 
   if (repository == NULL)
     {
@@ -227,7 +227,7 @@ gb_new_project_dialog__clone_worker (GTask        *task,
   workdir = ggit_repository_get_workdir (repository);
   g_task_return_pointer (task, g_object_ref (workdir), g_object_unref);
 
-  g_object_unref (repository);
+  g_clear_object (&repository);
 }
 
 static void
-- 
cgit v0.10.2

